gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpBasicAuthenticationHandler.java

```diff
@@ -112,14 +115,17 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
                             credentials.get(PROPERTY_USERNAME), address, authClassParts[authClassParts.length - 1]);
                 }
             } catch (AuthenticationException ae) {
-                sendError(ctx, msg);
+                sendError(ctx, msg, ae);
             }
         }
     }
 
-    private void sendError(final ChannelHandlerContext ctx, final Object msg) {
-        // Close the connection as soon as the error message is sent.
-        ctx.writeAndFlush(new DefaultFullHttpResponse(HTTP_1_1, UNAUTHORIZED)).addListener(ChannelFutureListener.CLOSE);
+    private void sendError(final ChannelHandlerContext ctx, final Object msg, final Throwable t) {
```

Comment:
Could we reuse HttpHandlerUtil's sendError here?

Reply:
reused